Fix existing channel reducer

lalexgap opened the pull request:
Fixes #545 

Previously the existing channel reducer assumed that both players would equally fund a channel. While this worked for RPS(since each player funds half of the total) but would not work in the general case.

This fixes the issue by using the allocation/destination of the app channel to figure out if the ledger channel needs to be topped up.

andrewgordstewart commented on packages/wallet/src/redux/protocols/existing-channel-funding/reducer.ts:
```diff
@@ -271,46 +267,36 @@ const waitForLedgerUpdateReducer = (
-  );
+  // We assume that destination/allocation are the same length and destination contains the same addresses
+  // Otherwise we shouldn't be in this protocol at all
+  for (let i = 0; i < proposedDestination.length; i++) {
```
This could be something like
```
return proposedDestination.some(address => {
  const existingIndex = latestCommitment.destination.indexOf(address);
  return bigNumberify(latestCommitment.allocation[existingIndex]).lt(proposedAllocation[i]);
}
``

lalexgap commented:
> Looks ok, but is there a test that exhibits the new behaviour? Ie. it would fail on master and would pass with this change?

I've changed the balances to be asymmetric that should catch the problem. Also added some basic top up tests.
